src/nvim/ex_docmd.c

```diff
@@ -5984,13 +5986,21 @@ char_u *get_user_cmd_addr_type(expand_T *xp, int idx)
 /*
  * Function given to ExpandGeneric() to obtain the list of user command names.
  */
-char_u *get_user_commands(expand_T *xp, int idx)
+char_u *get_user_commands(expand_T *xp FUNC_ATTR_UNUSED, int idx)
+  FUNC_ATTR_PURE FUNC_ATTR_WARN_UNUSED_RESULT
```

Comment:
This function doesn't seem to modify anything if `get_cmdline_type` is pure.

Reply:
modify what? This function is marked `FUNC_ATTR_PURE`, so it shouldn't modify anything.

Reply:
I marked it pure but the helper functions that it uses aren't explicitly declared as const/pure functions.